contracts/zero-ex/contracts/src/features/LiquidityProviderSwap.sol

```diff
@@ -0,0 +1,201 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+import "../errors/LibLiquidityProviderSwapRichErrors.sol";
+import "../fixins/FixinCommon.sol";
+import "../storage/LibLiquidityProviderSwapStorage.sol";
+import "./IFeature.sol";
+import "./ILiquidityProviderSwap.sol";
+import "./ITokenSpender.sol";
+
+
+interface IERC20Bridge {
+    /// @dev Transfers `amount` of the ERC20 `tokenAddress` from `from` to `to`.
+    /// @param tokenAddress The address of the ERC20 token to transfer.
+    /// @param from Address to transfer asset from.
+    /// @param to Address to transfer asset to.
+    /// @param amount Amount of asset to transfer.
+    /// @param bridgeData Arbitrary asset data needed by the bridge contract.
+    /// @return success The magic bytes `0xdc1600f3` if successful.
+    function bridgeTransferFrom(
+        address tokenAddress,
+        address from,
+        address to,
+        uint256 amount,
+        bytes calldata bridgeData
+    )
+        external
+        returns (bytes4 success);
+}
+
+contract LiquidityProviderSwap is
+    IFeature,
+    ILiquidityProviderSwap,
+    FixinCommon
+{
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibSafeMathV06 for uint256;
+    using LibRichErrorsV06 for bytes;
+
+    /// @dev Name of this feature.
+    string public constant override FEATURE_NAME = "LiquidityProviderSwap";
+    /// @dev Version of this feature.
+    uint256 public immutable override FEATURE_VERSION = _encodeVersion(1, 0, 0);
+
+    /// @dev ETH pseudo-token address.
+    address constant internal ETH_TOKEN_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    /// @dev The WETH contract address.
+    IEtherTokenV06 public immutable weth;
+
+    /// @dev Store the WETH address in an immutable.
+    /// @param weth_ The weth token.
+    constructor(IEtherTokenV06 weth_)
+        public
+        FixinCommon()
+    {
+        weth = weth_;
+    }
+
+    function sellToLiquidityProvider(
```

Comment:
I think we should probably emit an event in here for parity with the `FillQuoteTransformer`. I believe [this event ](https://github.com/0xProject/0x-monorepo/blob/development/contracts/zero-ex/contracts/src/features/TransformERC20.sol#L280) is used by our data integrations.